virtcontainers/kata_agent.go

```diff
@@ -234,6 +234,14 @@ func (k *kataAgent) configure(h hypervisor, id, sharePath string, builtin bool,
 		}
 	}
 
+	if builtin {
+		k.proxyBuiltIn = true
+	}
+
+	if fromGrpc {
```

Comment:
So when `fromGrpc == true`, the function is a no-op except for `proxyBuiltIn`. I'd prefer to just skip it from `NewVMFromGrpc` which is the sole caller of `configureFromGrpc`, and move the builtin proxy setting to `setProxy`.